vignettes/Rcpp-modules.Rmd

```diff
@@ -1059,74 +1136,183 @@ RCPP_MODULE(mod_vec) {
 
-When using \pkg{Rcpp} modules in a packages, the client package needs to
+As an example, consider a file called `yada.cpp` containing the following
+\proglang{C++} code:
+
+```{Rcpp eval=params$eval}
+#include <Rcpp.h>
+std::string hello() {
+    return "hello";
+}
+void bla() {
+    Rprintf("hello\\n");
+}
+void bla2( int x, double y) {
+    Rprintf("hello (x = %d, y = %5.2f)\\n", x, y);
+}
+
+class World {
+public:
+    World() : msg("hello") {}
+    void set(std::string msg) { this->msg = msg; }
+    std::string greet() { return msg; }
+
+private:
+    std::string msg;
+};
+
+RCPP_MODULE(yada){
+    using namespace Rcpp;
+
+    function("hello" , &hello);
+    function("bla"   , &bla);
+    function("bla2"  , &bla2);
+
+    class_<World>("World")
+    .constructor()
+    .method("greet", &World::greet)
+    .method("set",   &World::set)
+    ;
+}
+```
+
+```{r}
+sourceCpp('yada.cpp')
+```
+
+\proglang{C++} functions `hello`, `bla`, `bla2` and class `World` will be
+readily available in \proglang{R}:
+
+```{r, eval=params$eval, results=params$results}
+hello()
+bla()
+bla2(42, 0.42)
+w <- new(World)
+w$greet()
+w$set("hohoho")
+w$greet()
+```
+
+
+# Using modules in other packages {#sec:package}
+
+## Namespace import
+
+When using \textsl{Rcpp modules} in a packages, the client package needs to
 import \pkg{Rcpp}'s namespace. This is achieved by adding the
 following line to the `NAMESPACE` file.
 
+<!-- CHECK-PR: is this chunk needed at all? -->
 ```{r, echo=FALSE,eval=TRUE}
 options( prompt = " ", continue = " " )
 ```
 
-```{r, eval=FALSE}
+```{r}
 import(Rcpp)
 ```
 
 In some case we have found that explicitly naming a symbol can be preferable:
 
-```{r, eval=FALSE}
+```{r}
 import(Rcpp, evalCpp)
 ```
 
-## Load the module
+## Load the module in the namespace
+
+### Load the module content via loadModule {#sec:package-namespace-loadModule}
+
+Starting with release 0.9.11, the preferred way for loading a module directly
+into a package namespace is by calling the `loadModule()` function, which takes
+the module name as an argument and exposes the content of the module
+(\proglang{C++} functions and classes) as individual objects in the namespace.
+It can be placed in any `.R` file in the package. This is useful as it allows to
+load the module from the same file as some auxiliary \proglang{R} functions
+using the module.
 
-### Deprecated older method using loadRcppModules
+Consider a package \pkg{testmod} defining a module `yada` in the source file
+`src/yada.cpp`, with the same content as defined above in
+Section \ref{sec:modules-sourceCpp} above
 
-Note: This approach is deprecated as of Rcpp 0.12.5, and now triggers a warning
-message.  Eventually this function will be withdrawn.
+Then, `loadModule` is called in the package's \proglang{R} code to expose all
+\proglang{C++} functions and classes as objects `hello`, `bla`, `bla2`, `World`
+into the package namespace:
+
+```{r}
+loadModule("yada", TRUE)
```

Comment:
Perhaps switch this to a namespace function call?

```suggestion
Rcpp::loadModule("yada", TRUE)
```

Reply:
This is surely a valid point, although this approach does not seem to be used across other Rcpp vignettes.